### ISSUE TITLE
Pytorch implementation

### ISSUE THREAD

**Issue by animareversed** [ialhashim/DenseDepth#92]
Hi,
I tried to improve the Pytorch version of DenseDepth. This is a part of the code about the loss:
```
def image_gradients(x):
    l = x
    r = F.pad(x, [0, 1, 0, 0])[:, :, :, 1:]
    t = x
    b = F.pad(x, [0, 0, 0, 1])[:, :, 1:, :]

    dx, dy = torch.abs(r - l), torch.abs(b - t)
    dx[:, :, :, -1] = 0
    dy[:, :, -1, :] = 0

    return dx, dy 


           l_depth = self.l1_criterion(output, targets)

            ssim_module = SSIM(win_size=11, win_sigma=1.5, data_range=255.0, size_average=True, channel=1)
            l_ssim = torch.clamp(((1 - ssim_module(output, targets)) * 0.5), 0, 1)

            dy_true, dx_true = image_gradients(targets)
            dy_pred, dx_pred = image_gradients(output)
            l_edges = torch.mean(torch.abs(dy_pred - dy_true) + torch.abs(dx_pred - dx_true), dim=1)
            depth_losses['depth_loss'] = (1.0 * l_ssim) + (0.1 * l_depth) + (1.0 * torch.mean(l_edges))

```
but I don't know why the loss is not good, it's unstable (see the image below) when I add the edges loss, I used the function on top (image_gradients) and I compared it with tensorflow implementation of it and it seems they do the same thing. By any chance, do you know why this happen? 

![photo_2019-12-06_17-54-33](https://user-images.githubusercontent.com/55583897/70340657-7d30f700-1851-11ea-810f-64ae520dc7bc.jpg)

**Comment by ialhashim** [comment]
Not sure.. as I wrote at other issues I am not planning to support the PyTorch code, I only include it to help someone else brave enough to replicate our results.
